pandas/io/sql.py

```diff
@@ -257,15 +286,41 @@ def read_sql_table(
         raise ValueError(f"Table {table_name} not found", con)
 
 
+@overload
 def read_sql_query(
     sql,
     con,
     index_col=None,
     coerce_float=True,
     params=None,
     parse_dates=None,
-    chunksize=None,
-):
+    chunksize: int = 1,
+) -> DataFrame:
+    ...
+
+
+@overload
+def read_sql_query(
+    sql,
+    con,
+    index_col=None,
+    coerce_float=True,
+    params=None,
+    parse_dates=None,
+    chunksize: None = None,
+) -> Iterator[DataFrame]:
```

Comment:
are the return types of the two overloads switched? or have a misread the docstring?

Reply:
Yea looks like a copy / paste error? The other two funcs look correct.

lgtm outside of this issue